integreat_cms/static/src/js/analytics/hix-widget.ts

```diff
@@ -0,0 +1,149 @@
+
+  const chart: Chart = new Chart("hix-chart", {
+    type: "doughnut",
+    data: dummyData,
+    options: {
+      plugins: {
+        title: {
+          display: true,
+          text: ["HIX", "NA"],
+          position: "bottom",
+          padding: {
+            top: -150,
+          },
+          font: {
+            size: 24,
+          },
+        },
+      },
+      responsive: true,
+      cutout: "80%",
+    },
+  });
+
+  // Set listener for update button
+  document.getElementById("btn-update-hix-value").addEventListener("click", async (event) => {
+    document.getElementById("hix-loading")?.classList.remove("hidden");
+    event.preventDefault();
+    updateChart(chart, await getHixValue());
+  });
+
+  // Set listener, that checks, if tinyMCE content has changed to update the 
+  // HIX value status 
+  document.querySelectorAll("[data-content-changed]").forEach((element) => {
+    element.addEventListener("contentChanged", () => {
+      const updateButton = (document.getElementById("btn-update-hix-value") as HTMLSelectElement);
+      if (!getContent().trim()) {
+        updateButton.disabled = true; 
+        setHixLabelState("no-content");
+      } else {
+        updateButton.disabled = false; 
+        setHixLabelState("outdated"); 
+      }
+    });
+  });
+
+  // Delay the first request, so that tinyMCE can be initialized first
+  setTimeout(async () => {
+    if (!getContent().trim()) {
+      (document.getElementById("btn-update-hix-value") as HTMLSelectElement).disabled = true; 
+      setHixLabelState("no-content"); 
+      document.getElementById("hix-loading")?.classList.add("hidden");
+    } else {
+      updateChart(chart, await getHixValue());
+    }
+  }, 1000);
+
+});
+
+
+
+function updateChart(chart: Chart, value: number) {
+  const hixMaxValue = 20; 
+
+  chart.data.datasets.forEach((dataset) => {
+    // Set new data
+    dataset.data = [value, hixMaxValue - value];
+
+    // Set color based on HIX value
+    let backgroundColor;
+    if (value > 15) backgroundColor = 'rgb(50, 255, 50)';
+    else if (value > 7) backgroundColor = 'rgb(255, 255, 50)';
+    else backgroundColor = 'rgb(255, 50, 50)';
+    dataset.backgroundColor = [backgroundColor, 'rgb(220, 220, 220)'];
+  });
+  // Set Title to current HIX value
+  const roundedHixValue = (Math.round(value * 100) / 100).toString()
+  chart.options.plugins.title.text = ["HIX", roundedHixValue];
+  chart.update();
+};
+
+
+async function getHixValue() {
+  let result; 
+  await fetch("/ajax/get_hix_score/", {
+    method: "POST",
+    headers: {
+        "X-CSRFToken": getCsrfToken(),
+    },
+    body: JSON.stringify({
+        text: getContent(),
+    }),
+  }).then((response) => {
+    // Hide loading spinner
+    document.getElementById("hix-loading")?.classList.add("hidden");
+    return response.json();
+  }).then(
+    (json) => {
+      const labelState = json.error?"error":"updated";
+      setHixLabelState(labelState); 
+      result = json.score
```

Comment:
When an error occurs, I would hide the chart, since it looks a bit strange:
![Screenshot 2022-08-29 at 23-23-42 Integreat Redaktionssystem](https://user-images.githubusercontent.com/16021269/187301556-b10dce95-3771-46b4-aefd-cffcd94d89a3.png)